diffkemp/simpll/Transforms.cpp

```diff
@@ -113,13 +113,35 @@ std::vector<FunPair> simplifyModulesDiff(Config &config) {
             mam.getResult<FunctionAbstractionsGenerator>(*config.Second,
                                                          config.SecondFun));
 
+    StringRef FirstFunName = config.FirstFun->getName();
+    StringRef SecondFunName = config.SecondFun->getName();
+
     // Module passes
     PassManager<Module, AnalysisManager<Module, Function *>, Function *,
         Module *> mpm;
     mpm.addPass(RemoveUnusedReturnValuesPass {});
     mpm.run(*config.First, mam, config.FirstFun, config.Second.get());
     mpm.run(*config.Second, mam, config.SecondFun, config.First.get());
 
+    Function *NewFirstFun = nullptr, *NewSecondFun = nullptr;
+
+    // For the purpose of the comparision it is necessary to use the new
+    // function pointer in case the main function was replaced in order to
+    // compare the right functions.
+    if (config.FirstFun !=
+        config.First->getFunction(FirstFunName)) {
+        // Main function was replaced by a pass - fix the function pointer
+        NewFirstFun =
+                config.First->getFunction(FirstFunName);
+    }
+
+    if (config.SecondFun !=
+        config.Second->getFunction(SecondFunName)) {
+        // Main function was replaced by a pass - fix the function pointer
+        NewSecondFun =
+                config.Second->getFunction(SecondFunName);
+    }
+
```

Comment:
This seems a bit over-complicated. Wouldn't it be sufficient, after running the pass, to do `config.FirstFun = config.First.getFunction(FirstFunName)`?

Reply:
That was the original idea, but I think that the pointer to the original function is needed to get the source file path (somehow the `Fun->getSubprogram()` seems not to work with the new function - it's probably because it technically isn't a part of the original file), so I have to keep the pointer to it.

Reply:
Also (kind of unfortunately) while the getting of the source file needs the old function, the getting of its name needs the new function (because the old function doesn't have a name to avoid a conflict), that's why I keep both.

Reply:
Could you copy the pointer to the subprogram from the old function to the new one (there is a method `Fun->setSubprogram()`? If this would suffice, the code would be much simpler.

Reply:
That would be good if it suffices, because then I could also exclude the replacing of the main function in `ComparedFuns`.